dkg/pedersen/resharing_test.go

```diff
@@ -0,0 +1,142 @@
+	for i := 0; i < nOld; i++ {
+		actor, err := dkgsOld[i].Listen()
+		require.NoError(t, err)
+		actorsOld[i] = actor
+	}
+
+	_, err := actorsOld[1].Setup(fakeAuthority, thresholdOld)
+	require.NoError(t, err, "setting up the firs dkg was not successful")
+
+	// encrypt a message with the old committee public key. the new committee should be able to decrypt
+	// it successfully
+	message := []byte("Hello world")
+	K, C, remainder, err := actorsOld[0].Encrypt(message)
+	require.NoError(t, err, "encrypting the message was not successful")
+	require.Len(t, remainder, 0)
+
+	//setting up the second dkg
+	// nCommon is the number of nodes that are common between the new and the old committee
+	nCommon := 2
+	// the number of new added nodes. the new committee should have nCommon+nNew nodes in totatl
+	nNew := 40
+	thresholdNew := 42
+	minosNew := make([]mino.Mino, nNew+nCommon)
+	dkgsNew := make([]dkg.DKG, nNew+nCommon)
+	addrsNew := make([]mino.Address, nNew+nCommon)
+	//the first nCommon nodes of  committee are the same as the first nCommon nodes of the old committee
+	for i := 0; i < nCommon; i++ {
+		minosNew[i] = minosOld[i]
+		addrsNew[i] = minosOld[i].GetAddress()
+	}
+
+	pubkeysNew := make([]kyber.Point, len(minosNew))
+	// defining the address of the new nodes.
+	for i := 0; i < nNew; i++ {
+		minogrpc := minoch.MustCreate(minoManager, fmt.Sprintf("addr new %d", i))
+		minosNew[i+nCommon] = minogrpc
+		addrsNew[i+nCommon] = minogrpc.GetAddress()
+	}
+
+	// initializing the pedersen of the new nodes. the common nodes already have a pedersen
+	for i := 0; i < nNew; i++ {
+		dkg, pubkey := NewPedersen(minosNew[i+nCommon])
+		dkgsNew[i+nCommon] = dkg
+		pubkeysNew[i+nCommon] = pubkey
+	}
+	for i := 0; i < nCommon; i++ {
+		dkgsNew[i] = dkgsOld[i]
+		pubkeysNew[i] = pubkeysOld[i]
+	}
+
+	// initializing the actor of the new nodes. the common nodes already have an actor
+	actorsNew := make([]dkg.Actor, nNew+nCommon)
+	for i := 0; i < nCommon; i++ {
+		actorsNew[i] = actorsOld[i]
+	}
+	for i := 0; i < nNew; i++ {
+		actor, err := dkgsNew[i+nCommon].Listen()
+		require.NoError(t, err)
+		actorsNew[i+nCommon] = actor
+	}
+
+	// resharing the committee secret among the new committee
+	fakeAuthority = NewAuthority(addrsNew, pubkeysNew)
+	err = actorsOld[0].Reshare(fakeAuthority, thresholdNew)
+	require.NoError(t, err, "Resharing was not successful")
+
+	// comparing the public key of the old and the new committee
+	oldPubKey, err := actorsOld[0].GetPublicKey()
+	require.NoError(t, err)
+	for _, actorNew := range actorsNew {
+		newPubKey, err := actorNew.GetPublicKey()
+		// the public key should remain the same
+		require.NoError(t, err, "the public key should remain the same")
+		newPubKey.Equal(oldPubKey)
+		decrypted, err := actorNew.Decrypt(K, C)
+		require.NoError(t, err, "decryption was not successful")
+		require.Equal(t, message, decrypted, "the new committee should be able to decrypt the messages encrypted by the old committee")
```

Comment:
It would be nice to additionally create a 3rd new committee, that has nodes from the 2 previous ones